Remotion/Web/Development.WebTesting/Accessibility/IAccessibilityConfiguration.cs

```diff
@@ -0,0 +1,52 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using JetBrains.Annotations;
+
+namespace Remotion.Web.Development.WebTesting.Accessibility
+{
+  /// <summary>
+  /// Accessibility Analysis settings.
+  /// </summary>
+  /// <seealso cref="AccessibilityConfiguration"/>
+  public interface IAccessibilityConfiguration
+  {
+    /// <summary>Accessibility standard to use.</summary>
+    AccessibilityConformanceLevel ConformanceLevel { get; }
+
+    /// <summary>Whether IFrames are included.</summary>
+    bool IncludeIFrames { get; }
+
+    /// <summary>Timeout of IFrames in milliseconds.</summary>
+    TimeSpan IFrameTimeout { get; }
+
+    /// <summary>Whether ScrollToInitialPosition is enabled.</summary>
+    bool EnableScrollToInitialPosition { get; }
+
+    /// <summary>Whether AbsolutePaths are enabled.</summary>
+    bool EnableAbsolutePaths { get; }
+
+    /// <summary>Whether the axe result should contain XPaths.</summary>
+    bool EnableXPath { get; }
+
+    /// <summary>
+    /// Creates Json with specified options.
+    /// </summary>
+    [NotNull]
+    string GetJson ();
```

Comment:
Am not sure if GetJson and config-options should be on the same interface. Possibly, also not the same implemetnation.

Reply:
Perhaps `IAccessibilityConfiguration` could implement `ISerializable` and we'd just use a serializer to get the JSON representation?

Reply:
No, ISerializable is not meant to this. If, then you'd use a DataContract and the .NET JSON Serializer. Still would probably prefer to use separate Types, one the configuration and one the DTO to be json-serialized.

Reply:
Are you looking for some more specific `IJsonifiableAccessibilityConfiguration` or a more generic `IJsonSerializable` for re-motion?

Reply:
No to either. option 1) we just move the json-building to a different type, either an IAccessibleConfigurationJsonSerializer that acceepts the IAccessiblityConfiguration and returns a string, or we do it as a private code at the callsite of GetJson. Option 2) we make a DTO type, populate it and json-serialize, also at the callside of GetJson. It really depends on how and where we want to assert the json-string.

Reply:
In that case, to keep it simple I would probably just do it as private code at callside since the unit tests only cover the trivial scenarios and the interesting edge cases currently all need an integration test with real axe anyway

Reply:
Okay...?

As for what's needed: it's a matter of syntax. Even with out AXE / integration test we can verify that the complicated scenarios generate syntactically correct json. this alone might be worth doing dedicated tests for. Otherwise, we depend on not making mistakes and if there's mistakes, the AXE / JS-engine giving us at least error detection if not helpful details.